django/http/multipartparser.py

```diff
@@ -321,6 +321,10 @@ def sanitize_file_name(self, file_name):
         file_name = file_name.rsplit('/')[-1]
         file_name = file_name.rsplit('\\')[-1]
 
+        # Remove all non-printable characters from file_name
+        _ = list(char for char in file_name if char.isprintable())
+        file_name = ''.join(_)
```

Comment:
A temporary variable is not necessary:
```suggestion
        # Remove all non-printable characters.
        file_name = ''.join([char for char in file_name if char.isprintable()])
```
Also, it should be faster to check with `string.printable`:
```python
file_name = ''.join([x for x in file_name if x in string.printable])
```
```
$ python3 -m timeit 'import random; x = "".join([chr(random.randint(1, 10000)) for _ in range(10000)]); "".join([char for char in x if char.isprintable()])'
50 loops, best of 5: 7.03 msec per loop

$ python3 -m timeit 'import random; import string; x = "".join([chr(random.randint(1, 10000)) for _ in range(10000)]); "".join([char for char in x if char in string.printable])'
50 loops, best of 5: 6.76 msec per loop
```

Reply:
The main problem with `string.printable` is that it does not include any unicode characters than the standard Latin alphabet, not even accented characters. Replacing `str.isprintable()` for `string.printable` immediately makes 5 tests fail that make sure Unicode characters in the file name are processed correctly.